### DISCUSSION TITLE
Use of mixin tag to Model file

### DISCUSSION THREAD

**Discussion by KarinaRashchynskaya** [Soyhuce/next-ide-helper#104] [Q&A] [answered]
Hello,

I would like to ask, if there is a way to override the Models file, so the whole generated docblock, which is produced by the command :  `next-ide-helper:models`, can be moved to `_ide_models.php` file and replaced by something like that :

```
/**
 * …
 * @mixin IdeHelperArticle
 */
 
class Article extends Model
{}
```
 
 **_ide_models.php**
```
/**
 * @property int $id
 * @property string $title
 */
class __IdeHelperArticle
{}
```

In this way, we can keep our Model class clearer, as only one tag will be added to file and the rest will be written in (`_ide_models.php`).

So, is there any feature like the above now or will it be in the future?

Thank you.

**Comment by KarinaRashchynskaya** [comment]
Hello @bastien-phi,

Any news on this?

Thank you

**Comment by bastien-phi** [comment]
Sorry about my late answer. A lot of client work currently...

It is not possible yet to extract such a mixin nor scheduled but I'm completely opened to a PR with this feature.

**Comment by haringsrob** [comment]
I did pr this in the past: https://github.com/Soyhuce/next-ide-helper/pull/57

**Reply by haringsrob** [reply]
Looking into fixing that.

**Comment by bastien-phi** [comment, marked as answer]
Finally had some time for this. Try version 2.0 !

You'll find some information in the [doc](https://github.com/Soyhuce/next-ide-helper/tree/2.0.0?tab=readme-ov-file#mixin-style-helper).
